src/lib/storage/base_segment.hpp

```diff
@@ -32,6 +34,11 @@ class BaseSegment : private Noncopyable {
   // such as strings who memory usage is implementation defined
   virtual size_t estimate_memory_usage() const = 0;
 
+  SegmentAccessStatistics_T& access_statistics() const;
```

Comment:
There was a comment on here and you replied

> I don't want anybody to be able to change th counter externally.

Where did that go?

Anyway - you DO allow people to modify the counter as you do not return a const reference.